content/iOS/Practices/iOS-practices.md

```diff
@@ -0,0 +1,116 @@
+
+## Architecture Patterns
+
+Be very mindful of the pattern you are gonna follow in your new project and stick with it! 
+
+There's no right answer on what pattern to follow, it depends a lot on your project. I won't be enumerating every option as there are so many, and new ones keep being invented every day according to the new challenges we face as time passes, and we start seeing flaws in the existent ones.
+So, let's stick to the more common ones for now. 
+
+### MVC
+
+Like many others, at first I started coding in MVC, one of the simpler ones. 
+This pattern divides responsibilities into three class types: Models, Views, and Controllers. 
+Models deals with data logic, Views display the UI and the Controllers handles communication between the two, navigation, and business logic.  
+
+The problem with this pattern is that a lot of the responsibilities fall in the controllers' job, making them huge and ugly beasts that can be a real pain to maintain, test, and reuse in the future. 
+
+### MVVM
+
+This pattern divides responsibilities within Models, Views, and View Models, and is one of the most popular.
+Business logic and communication between Models and Views are now responsibility of the ViewModel, leaving the Controller job to only handle user interaction, displaying the digested information the View Model provides in Views, and handling navigation. 
+
+This extra separation of concerns allows us to keep a cleaner code base as well as facilitating unit testing.
+
+There are a bunch of variations of this pattern, the most popular ones include a Coordinator that would handle navigation and flow of the app relieving view controllers from this work.  
+
+For an example of this implementation you can check out [Rootstrap's iOS base](https://github.com/rootstrap/ios-base) a very helpful iOS project template to let you jump into your new project quickly. Check this [article](https://www.rootstrap.com/blog/2019/10/25/whats-great-about-our-ios-base/) to find out more about why we did that and how helpful it has been for us.  
+
+## Resource Management 
+
+As we mentioned before we want to have a folder in the project's root for the resources our app will need, there are many types of resources so its helpful to organize them in different folders according to their type, for example, you may have: property lists, images, gifs, JSONs... and the list goes on.
+
+For images, Xcode provides a very helpful resource bundle named xassets. 
+Here we'll store the images our app uses. If your app has a fancy UI and uses loads of them is good to have them grouped in folders that describe their use. 
+
+Images may be heavy, and we want our apps to be as small in size as possible, so if you find yourself having to add the same image in different colors its recommended to use them as templates and color them in code instead of adding different versions of the same image. 
```

Comment:
```suggestion
Images may be heavy, and we want our apps to be as small in size as possible, so if you find yourself having to add the same image in different colors it's recommended to use them as templates and color them in code instead of adding different versions of the same image. 
```

This is actually really good advice! For some time (longer than I would like to admit) I was completely unaware that it was possible